client/src/components/RelatedItems/RelatedItemsHooks.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import _ from 'underscore';
+import './style.css';
+import Related from './Related.jsx';
+import Comparison from './Comparison.jsx';
+
+export const CombinedAPIDetails = React.createContext();
+
+function RelatedItemsHooks() {
+  const [productId, setProductId] = useState(38013);
+  const [relatedItems, setRelatedItems] = useState([]);
+  const [relatedItemsImg, setRelatedItemsImg] = useState([]);
+  const [relatedItemsDetails, setRelatedItemsDetails] = useState([]);
+  const [combinedAPIDetails, setCombinedAPIDetails] = useState([]);
+  const [outfit, setOutfit] = useState([]);
+
+  useEffect(() => {
+    axios
+      .get(`/products/${productId}/related`)
+      .then((res) => setRelatedItems(res.data))
+      .catch((err) => console.error(err));
+  }, [productId]);
+
+  useEffect(() => {
+    Promise.all(
+      relatedItems.map((product) =>
+        axios
+          .get(`/products/${product}/styles`)
+          .then((response) => response.data)
+          .catch((err) => <div>There was an API error: {err}</div>)
+      )
+    )
+      .then((response) => setRelatedItemsImg(response))
+      .catch((err) => console.error(err));
+  }, [relatedItems]);
+
+  useEffect(() => {
+    Promise.all(
+      relatedItems.map((product) =>
+        axios
+          .get(`/products/${product}`)
+          .then((response) => response.data)
+          .catch((err) => console.error(err))
+      )
+    ).then((res) => setRelatedItemsDetails(res));
+  }, [relatedItems]);
+
+  useEffect(() => {
+    const accArr = []
+    for (let i = 0; i < relatedItems.length; i++ ) {
+      accArr.push(_.extend(relatedItemsImg[i], relatedItemsDetails[i]));
```

Comment:
Clever!

For future reference: You don't need an external library to do this. 

In ES5 syntax: `Object.assign(relatedItemsImg[i], relatedItemsDetails[i])`.

In ES6 syntax: `{ ...relatedItemsImg[i], ...relatedItemsDetails[i] }`